### PR TITLE
"Main" branch clean-up

### PR DESCRIPTION
**Pull request by kpyszkowski**
Ref: #1 
Depends on: #7 #8 #10 #11 

This pull request aggregates each partial clean-up PR (mentioned above). It's introduced to reduce complexity of merge conflicts and allow continuous work without blocking while waiting for partial PRs to be merged.
Moreover it's a place for general cleanup commits that couldn't be addressed to any of the PRs.

### CODE REVIEW

**Comment by kpyszkowski** [comment]
@michalsmiarowski 
I've spotted an error that doesn't occur on branches #7, #8, #10, #11 but in this one. I might have messed something with resolving conflicts but I can't find the direct origin of the issue. 
When you disconnect the wallet using a button (calling `disconnect` function returned by `useWeb3React` hook) after few seconds it throws an error from the screenshot. in fact it breaks in [this line](https://github.com/threshold-network/bitcoin-on-base/blob/main/src/components/tBTC/BridgeActivity.tsx#L113) because as you disconnect the wallet, the `account` value from `useWeb3React` hook becomes `undefined`.
I'm curious how this error didn't occur earlier because from the code perspective that is "expectable" behavior.

Could you investigate it? I’ve already hit the wall, all my efforts in this matter are pointless 😩 

image

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
First round of review

File: src/store/index.ts
Comment (michalsmiarowski) [reply src/store/index.ts:25]: I've made it non-blocking, because I think I can address it in a separate PR too to not distract you from the UI work and to not block this PR.

File: src/store/index.ts
Comment (michalsmiarowski) [reply src/store/index.ts:23]: I've made it non-blocking, because I think I can address it in a separate PR too to not distract you from the UI work and to not block this PR.

File: src/App.tsx
Comment (kpyszkowski) [reply src/App.tsx]: Ref commit: 920b31dc

File: src/components/tBTC/BridgeActivity.tsx
Comment (michalsmiarowski) [inline src/components/tBTC/BridgeActivity.tsx:97]: Doesn't `!!` return true for and empty array? I think it does.

In that case it would be best to return `isBridgeHistoryEmpty` to the bridge activity context and use it instead of `!!data`.

File: src/components/tBTC/BridgeActivity.tsx
Comment (michalsmiarowski) [inline src/components/tBTC/BridgeActivity.tsx:70]: Non-blocking:

`isWalletConnected` is not really related to `BridgeActivity` so I'm wondering if it's worth placing it in the context? Maybe we could just call `const { active } = useWeb3React()` where it's needed?

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Left some comments to look at before the merge

File: src/components/tBTC/BridgeActivity.tsx
Comment (michalsmiarowski) [reply src/components/tBTC/BridgeActivity.tsx:192]: Aditional note:

Based on https://github.com/threshold-network/bitcoin-on-base/pull/12#discussion_r1450489579 and https://github.com/threshold-network/bitcoin-on-base/pull/12#discussion_r1450491804 I would actually implement it like this:

```
  const { active } = useWeb3React()
  (...)
  const shouldRenderEmptyState = !active || (isBridgeHistoryEmpty && !isFetching)
```

Comment (kpyszkowski) [reply src/components/tBTC/BridgeActivity.tsx]: Ref commit: 78f10f1e

File: src/components/tBTC/BridgeActivity.tsx
Comment (kpyszkowski) [reply src/components/tBTC/BridgeActivity.tsx]: Ref commit: cfb65f4d

File: src/components/tBTC/BridgeActivity.tsx
Comment (kpyszkowski) [reply src/components/tBTC/BridgeActivity.tsx]: Ref commit: cfb65f4d

File: src/components/tBTC/BridgeActivity.tsx
Comment (michalsmiarowski) [reply src/components/tBTC/BridgeActivity.tsx]: Not resolved yet. See https://github.com/threshold-network/bitcoin-on-base/pull/12#discussion_r1452161131

File: src/components/tBTC/BridgeActivity.tsx
Comment (kpyszkowski) [reply src/components/tBTC/BridgeActivity.tsx]: Ref commit: f568babf

**Review by michalsmiarowski — APPROVED** [review]
LGTM 🔥

**Pull request merged**
